server/lib/src/plugins/nameupdate.rs

```diff
@@ -0,0 +1,240 @@
+    use crate::prelude::uuid;
+    use crate::value::Value;
+
+    #[test]
+    fn name_purge_and_set_1() {
+        // Add another uuid to a type
+        let ea = entry_init!(
+            ("class", Value::new_class("account")),
+            ("class", Value::new_class("posixaccount")),
+            ("name", Value::new_iname("old_name")),
+            (
+                "uuid",
+                Value::Uuid(uuid!("d2b496bd-8493-47b7-8142-f568b5cf47ee"))
+            ),
+            ("namehistory", Value::new_iname("old_old_name")),
+            ("description", Value::new_utf8s("testperson")),
+            ("displayname", Value::new_utf8s("old name person"))
+        );
+        let preload = vec![ea];
+        run_modify_test!(
+            Ok(()),
+            preload,
+            filter!(f_eq("name", PartialValue::new_iname("old_name"))),
+            modlist!([
+                m_purge("name"),
+                m_pres("name", &Value::new_iname("new_name_1"))
+            ]),
+            None,
+            |_| {},
+            |qs: &mut QueryServerWriteTransaction| {
+                let e = qs
+                    .internal_search_uuid(uuid!("d2b496bd-8493-47b7-8142-f568b5cf47ee"))
+                    .expect("failed to get entry");
+                dbg!(e.get_ava());
+                let c = e
+                    .get_ava_set("namehistory")
+                    .expect("failed to get primary cred.");
+                return assert!(c.contains(&PartialValue::new_iname(&"old_name")));
+            }
+        );
+    }
+    #[test]
+    fn name_purge_and_set_2() {
+        // Add another uuid to a type
+        let ea = entry_init!(
+            ("class", Value::new_class("account")),
+            ("class", Value::new_class("posixaccount")),
+            ("name", Value::new_iname("old_name")),
+            (
+                "uuid",
+                Value::Uuid(uuid!("d2b496bd-8493-47b7-8142-f568b5cf47ee"))
+            ),
+            ("description", Value::new_utf8s("testperson")),
+            ("displayname", Value::new_utf8s("old name person"))
+        );
+        let preload = vec![ea];
+        run_modify_test!(
+            Ok(()),
+            preload,
+            filter!(f_eq("name", PartialValue::new_iname("old_name"))),
+            modlist!([
+                m_purge("name"),
+                m_pres("name", &Value::new_iname("new_name_1"))
+            ]),
+            None,
+            |_| {},
+            |qs: &mut QueryServerWriteTransaction| {
+                let e = qs
+                    .internal_search_uuid(uuid!("d2b496bd-8493-47b7-8142-f568b5cf47ee"))
+                    .expect("failed to get entry");
+                dbg!(e.get_ava());
+                let c = e
+                    .get_ava_set("namehistory")
+                    .expect("failed to get primary cred.");
+                return assert!(c.contains(&PartialValue::new_iname(&"old_name")));
+            }
+        );
+    }
```

Comment:
Should there be another test to check what happens if the user changes their name to the *same* value as it currently is?